src/kOS/Suffixed/VectorRenderer.cs

```diff
@@ -74,7 +75,13 @@ public void ScopeLost()
             // (Note that if I didn't do this,
             // then as far as C# thinks, I wouldn't be orphaned because
             // UpdateHandler is holding a reference to me.)
-            SetShow(false);
+
+            // Store the information about lost scope to be checked in the KOSUpdate method
+            // because Unity complains if we try to set visual element visibility from
+            // outside of the main thread.  That isn't a problem most of the time, but the
+            // finalizer method called by the GC (see Variable.cs ~Variable method) is
+            // apparently called from another thread, or at least Unity thinks it is.
+            scopeLost = true;
```

Comment:
I hadn't really thought about that, but yes, we could modify `ScopeLost` to do nothing, and check `LinkCount` instead.  It's more clean than adding another variable.